### DISCUSSION TITLE
How to use installed parser for a file?

### DISCUSSION THREAD

**Discussion by ildar-shaymukhametov** [nvim-treesitter/nvim-treesitter#6363] [General]
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/74532770/dd09717e-a2f0-4d75-a512-ef6a68e8aa91)
I have neither a parser nor lsp for idris programming language (.idr)
But I want for idris files to use lua parser

vim.treesitter.language.register('lua', 'idris')
vim.treesitter.language.register('lua', 'idr')

 These do not work
 How to make treesitter use lua parser for idris files (.idr)?

**Comment by clason** [comment]
You need to explicitly start the highlighter. Easiest is to just add an autocommand in your `init.lua`:
```lua
vim.api.nvim_create_autocmd('FileType', {
  pattern = 'idris',
  callback = function(args) vim.treesitter.start(args.buf, 'lua') end,
})
```

**Reply by ildar-shaymukhametov** [reply]
![image](https://github.com/nvim-treesitter/nvim-treesitter/assets/74532770/2e821314-2e20-4663-a867-033adef62e3e)
Didn't work
I updated init.lua
Closed neovim
Opened neovim again
Also I tried different values for the field "pattern": '*.idr', '.idr', 'idr'

**Reply by clason** [reply]
Put it directly in your `init.lua`; this has nothing to do with nvim-treesitter (or whatever plugin manager you use).

**Reply by ildar-shaymukhametov** [reply]
I did that, but same result
It works only if I do :set filetype=idris
It cannot understand that *.idr files are of type idris
Maybe I need to associate them somehow?

**Reply by clason** [reply]
Yes, that's a different issue. You need to use `vim.filetype.add` to register the filetype. (Please read the Neovim documentation.)

**Reply by ildar-shaymukhametov** [reply]
Ok, thanks!
